### ISSUE TITLE
Slack notification not sent when pod is restarted

### ISSUE THREAD

**Issue by dnoland1** [rancher/rancher#20339]
**What kind of request is this (question/bug/enhancement/feature request):**
bug

**Steps to reproduce (least amount of steps as possible):**
1. Create a slack notifier. I clicked the "Test" button and got: 
<img width="346" alt="image" src="https://user-images.githubusercontent.com/468462/58039724-74144e80-7ae8-11e9-9486-729196b612dd.png">
2. Created a one node custom cluster on AWS using t3a.medium, ros 1.5.1.
3. Turned on monitoring for the cluster.
4. Turned on monitoring for the default project.
5. Deployed a workload in the default project called "nginxtest" using the "nginx" image. Confirmed workload was running.
6. Created a new alert in the default project to trigger when the pod is restarted: <img width="1242" alt="image" src="https://user-images.githubusercontent.com/468462/58039577-ffd9ab00-7ae7-11e9-92c7-c9ff7e84d34b.png">
7. Edited the alert group and set my slack notifier as the notifier.
8. On the worker node, ran a "docker ps | grep nginxtest", then with the id, ran "docker inspect <id> | grep -i PID", then ran "sudo kill -9 <pid>". This caused the pod to restart: 
<img width="1239" alt="image" src="https://user-images.githubusercontent.com/468462/58039995-277d4300-7ae9-11e9-9012-70d7bef9b7be.png">

**Result:**
No slack notification was received. I would expect to receive one since I set an alert that should be triggered if the pod was restarted.

**Other details that may be helpful:**

**Environment information**
- Rancher version (`rancher/rancher`/`rancher/server` image tag or shown bottom left in the UI):
v2.2.1
- Installation option (single install/HA):
single install

**Cluster information**
- Cluster type (Hosted/Infrastructure Provider/Custom/Imported):
Custom
- Machine type (cloud/VM/metal) and specifications (CPU/memory):
Cloud - AWS t3a.medium
- Kubernetes version (use `kubectl version`):

```
$ kubectl version
Client Version: version.Info{Major:"1", Minor:"12", GitVersion:"v1.12.7", GitCommit:"6f482974b76db3f1e0f5d24605a9d1d38fad9a2b", GitTreeState:"clean", BuildDate:"2019-03-25T02:52:13Z", GoVersion:"go1.10.8", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"13", GitVersion:"v1.13.5", GitCommit:"2166946f41b36dea2c4626f90a77706f426cdea2", GitTreeState:"clean", BuildDate:"2019-03-25T15:19:22Z", GoVersion:"go1.11.5", Compiler:"gc", Platform:"linux/amd64"}
```

- Docker version (use `docker version`):

```
[rancher@ip-10-0-0-10 ~]$ docker version
Client:
 Version:           18.06.1-ce
 API version:       1.38
 Go version:        go1.10.3
 Git commit:        e68fc7a
 Built:             Tue Aug 21 17:20:43 2018
 OS/Arch:           linux/amd64
 Experimental:      false

Server:
 Engine:
  Version:          18.06.1-ce
  API version:      1.38 (minimum version 1.12)
  Go version:       go1.10.3
  Git commit:       e68fc7a
  Built:            Tue Aug 21 17:28:38 2018
  OS/Arch:          linux/amd64
  Experimental:     false
```

**Comment by aiwantaozi** [comment]
Could you check the the the pod -> status -> container status -> restart count in the API ?

**Comment by dnoland1** [comment]
@aiwantaozi Yes, see restartCount: 1 in API. Here's the full dump:
```json
{

    "annotations": {
        "cattle.io/timestamp": "2019-05-23T20:55:42Z",
        "cni.projectcalico.org/podIP": "10.42.0.12/32"
    },
    "baseType": "pod",
    "containers": [
        {
            "allowPrivilegeEscalation": false,
            "exitCode": null,
            "image": "nginx",
            "imagePullPolicy": "Always",
            "initContainer": false,
            "name": "nginxtest",
            "privileged": false,
            "procMount": "Default",
            "readOnly": false,
            "resources": {
                "type": "/v3/project/schemas/resourceRequirements"
            },
            "restartCount": 1,
            "runAsNonRoot": false,
            "state": "running",
            "stdin": true,
            "stdinOnce": false,
            "terminationMessagePath": "/dev/termination-log",
            "terminationMessagePolicy": "File",
            "transitioning": "no",
            "transitioningMessage": "",
            "tty": true,
            "type": "/v3/project/schemas/container",
            "volumeMounts": [
                {
                    "mountPath": "/var/run/secrets/kubernetes.io/serviceaccount",
                    "name": "default-token-zvcf7",
                    "readOnly": true,
                    "type": "/v3/project/schemas/volumeMount"
                }
            ]
        }
    ],
    "created": "2019-05-23T20:55:44Z",
    "createdTS": 1558644944000,
    "creatorId": null,
    "dnsPolicy": "ClusterFirst",
    "hostIPC": false,
    "hostNetwork": false,
    "hostPID": false,
    "id": "default:nginxtest-fbcfc469f-wh97q",
    "labels": {
        "pod-template-hash": "fbcfc469f",
        "workload.user.cattle.io/workloadselector": "deployment-default-nginxtest"
    },
    "links": {
        "remove": "…/v3/project/c-9ffvr:p-glhr2/pods/default:nginxtest-fbcfc469f-wh97q",
        "self": "…/v3/project/c-9ffvr:p-glhr2/pods/default:nginxtest-fbcfc469f-wh97q",
        "update": "…/v3/project/c-9ffvr:p-glhr2/pods/default:nginxtest-fbcfc469f-wh97q",
        "yaml": "…/v3/project/c-9ffvr:p-glhr2/pods/default:nginxtest-fbcfc469f-wh97q/yaml"
    },
    "name": "nginxtest-fbcfc469f-wh97q",
    "namespaceId": "default",
    "nodeId": "c-9ffvr:m-db4d08fca483",
    "ownerReferences": [
        {
            "apiVersion": "apps/v1",
            "blockOwnerDeletion": true,
            "controller": true,
            "kind": "ReplicaSet",
            "name": "nginxtest-fbcfc469f",
            "type": "/v3/project/schemas/ownerReference",
            "uid": "22175d51-7d9d-11e9-b2cf-0a7549f22dac"
        }
    ],
    "priority": 0,
    "projectId": "c-9ffvr:p-glhr2",
    "restartPolicy": "Always",
    "schedulerName": "default-scheduler",
    "scheduling": {
        "tolerate": [ 2 items
            {
                "effect": "NoExecute",
                "key": "node.kubernetes.io/not-ready",
                "operator": "Exists",
                "tolerationSeconds": 300,
                "type": "/v3/project/schemas/toleration"
            },
            {
                "effect": "NoExecute",
                "key": "node.kubernetes.io/unreachable",
                "operator": "Exists",
                "tolerationSeconds": 300,
                "type": "/v3/project/schemas/toleration"
            }
        ]
    },
    "serviceAccountName": "default",
    "state": "running",
    "status": {
        "conditions": [ 4 items
            {
                "lastProbeTime": null,
                "lastTransitionTime": "2019-05-23T20:55:44Z",
                "lastTransitionTimeTS": 1558644944000,
                "status": "True",
                "type": "Initialized"
            },
            {
                "lastProbeTime": null,
                "lastTransitionTime": "2019-05-23T20:58:49Z",
                "lastTransitionTimeTS": 1558645129000,
                "status": "True",
                "type": "Ready"
            },
            {
                "lastProbeTime": null,
                "lastTransitionTime": "2019-05-23T20:58:49Z",
                "lastTransitionTimeTS": 1558645129000,
                "status": "True",
                "type": "ContainersReady"
            },
            {
                "lastProbeTime": null,
                "lastTransitionTime": "2019-05-23T20:55:44Z",
                "lastTransitionTimeTS": 1558644944000,
                "status": "True",
                "type": "PodScheduled"
            }
        ],
        "containerStatuses": [
            {
                "containerID": "docker://3ac433502d103fd36bc84198c8182c5fb2a6e1d903746244f34d78fb55a0d845",
                "image": "nginx:latest",
                "imageID": "docker-pullable://nginx@sha256:23b4dcdf0d34d4a129755fc6f52e1c6e23bb34ea011b315d87e193033bcd1b68",
                "lastState": {
                    "terminated": {
                        "containerID": "docker://3271df6a15a6a97e3c5f1a710747185704cc351486a0a649aef485378c4a72c0",
                        "exitCode": 137,
                        "finishedAt": "2019-05-23T20:58:47Z",
                        "finishedAtTS": 1558645127000,
                        "reason": "Error",
                        "signal": 0,
                        "startedAt": "2019-05-23T20:55:52Z",
                        "startedAtTS": 1558644952000,
                        "type": "/v3/project/schemas/containerStateTerminated"
                    },
                    "type": "/v3/project/schemas/containerState"
                },
                "name": "nginxtest",
                "ready": true,
                "restartCount": 1,
                "state": {
                    "running": {
                        "startedAt": "2019-05-23T20:58:49Z",
                        "startedAtTS": 1558645129000,
                        "type": "/v3/project/schemas/containerStateRunning"
                    },
                    "type": "/v3/project/schemas/containerState"
                },
                "type": "/v3/project/schemas/containerStatus"
            }
        ],
        "nodeIp": "10.0.0.10",
        "phase": "Running",
        "podIp": "10.42.0.12",
        "qosClass": "BestEffort",
        "startTime": "2019-05-23T20:55:44Z",
        "startTimeTS": 1558644944000,
        "type": "/v3/project/schemas/podStatus"
    },
    "terminationGracePeriodSeconds": 30,
    "transitioning": "no",
    "transitioningMessage": "",
    "type": "pod",
    "uuid": "221d24d8-7d9d-11e9-b2cf-0a7549f22dac",
    "volumes": [
        {
            "name": "default-token-zvcf7",
            "secret": {
                "defaultMode": 420,
                "secretName": "default-token-zvcf7",
                "type": "/v3/project/schemas/secretVolumeSource"
            },
            "type": "/v3/project/schemas/volume"
        }
    ],
    "workloadId": "deployment:default:nginxtest"

}
```

**Comment by jiaqiluo** [comment]
Following the steps in the issue, the bug is reproduced in Rancher:v2.2.6 and the bug fix is validated in Rancher:master-head 895f8dd1

Results:

<img width="1444" alt="Screen Shot 2019-07-26 at 1 46 09 PM" src="https://user-images.githubusercontent.com/6218999/61981184-c58bfe80-afad-11e9-9de4-bff46fa25c48.png">

![Screen Shot 2019-07-26 at 1 46 19 PM](https://user-images.githubusercontent.com/6218999/61981185-c58bfe80-afad-11e9-8f63-e85ace84b0ac.png)
